06_create_cluster.sh

```diff
@@ -36,4 +36,19 @@ if [[ ! -z "${ENABLE_LOCAL_REGISTRY}" ]]; then
       --type merge --patch "{\"spec\":{\"samplesRegistry\":\"${LOCAL_REGISTRY_DNS_NAME}\"}}"
 fi
 
+if [[ ! -z "${APPLY_EXTRA_WORKERS}" ]]; then
+    if [[ ${NUM_EXTRA_WORKERS} -ne 0 && -f "${OCP_DIR}/extra_host_manifests.yaml" ]]; then
+        oc apply -f "${OCP_DIR}/extra_host_manifests.yaml"
+
+        set +x
+        while oc get baremetalhost -n openshift-machine-api -o json | jq '.items[].status.poweredOn' | grep -q "false"; do
+            sleep 30
```

Comment:
If we're going to check the power state here, it should be [${EXTRA_WORKERS_ONLINE_STATUS}](https://github.com/openshift-metal3/dev-scripts/blob/master/ocp_install_env.sh#L284)

But I don't think that's necessarily going to be reliable in all cases, since it's possible that the workers in some other state.

Instead, I'd suggest checking for the extra hosts explicitly, e.g something like:

```
for h in $(jq -r '.[].name' ${EXTRA_BAREMETALHOSTS_FILE}); do
  while ! oc get baremetalhost -n openshift-machine-api $h 2>/dev/null; do
    echo "Waiting for $h"
    sleep 1
  done
done
```

Also note the reduced sleep time - the apply should happen pretty fast so I don't think it's necessary to wait 30s?

Reply:
Although apply is pretty fast, host stays in inspecting state for a while and if the scale in/out test runs in this phase(pretty low probability but it might create flakiness), test fails.

Reply:
Ack, good point - we could add a check for a specific provisioning state then (just checking the power status won't be enough), and we may want to make that configurable since e.g in https://github.com/openshift-metal3/dev-scripts/pull/1193 I'm adding options which will result in the BMH state being `provisioned` vs just ready when no image is specified?

Reply:
re the sleep time perhaps we can compromise and use 5s or something - my motivation is I don't want to wait 30s unnecessarily in the live-iso flow (where inspection may be disabled).

Reply:
is it meaningful to create a new environment variable like `APPLY_EXTRA_WORKERS_WAIT_STATE` default is `ready`?, or maybe we can continue with your proposal above and if we see any flakiness, we add state check later.

Reply:
> Although apply is pretty fast, host stays in inspecting state for a while and if the scale in/out test runs in this phase(pretty low probability but it might create flakiness), test fails.

I don't think it's correct to push at this level a test requirement. If the test requires a waiting condition (and that's fully reasonable), it must be in the test code, and not in the setup (since it could change in future, or be completely replaced by a different setup mechanism). So I guess that 5s delay it's acceptable as a starting point

Reply:
I partially disagree. Test tries to scale in and out without extra paying attention of aws, gcp or dev-scripts. I agree, this is not dev-scripts mission to be implemented in the way of specific case of this test. If @hardys will not need in live-iso flow as a general need, I think we can move forward with 5s delay.

Reply:
> I don't think it's correct to push at this level a test requirement. If the test requires a waiting condition (and that's fully reasonable), it must be in the test code, and not in the setup

Would those tests support checking the BMH state as a precursor to scaling up the MachineSet, or are they completely platform agnostic?  I was assuming the latter, hence thinking we would check the BMH state here.

Another option is just don't check the BMH state at all, then the test will still eventually fail when the new Machine fails to move to `Provisioned` - it'll just take longer.